Content.Server/TapeRecorder/CassetteTapeSystem.cs

```diff
@@ -0,0 +1,134 @@
+using Content.Shared.Popups;
+using Content.Shared.Tag;
+using Content.Shared.Tools.Components;
+using Content.Shared.Weapons.Melee;
+using Robust.Server.GameObjects;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Server.TapeRecorder
+{
+    public sealed class CassetteTapeSystem : EntitySystem
+    {
+        [Dependency] private readonly TagSystem _tagSystem = default!;
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+
+        /// <inheritdoc/>
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<CassetteTapeComponent, DamageChangedEvent>(OnDamagedChanged);
+            SubscribeLocalEvent<CassetteTapeComponent, InteractUsingEvent>(OnInteractUsing);
+            SubscribeLocalEvent<CassetteTapeComponent, SpoolingCompleteEvent>(OnSpoolingComplete);
+            SubscribeLocalEvent<CassetteTapeComponent, SpoolingCancelEvent>(OnSpoolingCancel);
+        }
+
+
+        /// <summary>
+        /// Handles "fixing" the cassette tape by screwing it back in with a screwdriver (ADD SUPPORT FOR PENCILS WHEN THOSE ARE ADDED)
+        /// </summary>
+        private void OnInteractUsing(EntityUid uid, CassetteTapeComponent component, InteractUsingEvent args)
+        {
+            if (!component.Unspooled || component.CancelToken != null)
+                return;
+            if (!TryComp<ToolComponent>(args.Used, out var toolComponent))
+                return;
+            if (!toolComponent.Qualities.Contains("Screwing"))
+                return;
+
+            component.CancelToken = new CancellationTokenSource();
+
+            var doAfterEventArgs = new DoAfterEventArgs(args.User, 7, component.CancelToken.Token, component.Owner)
+            {
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                BreakOnDamage = true,
+                BreakOnStun = true,
+                NeedHand = true,
+                TargetFinishedEvent = new SpoolingCompleteEvent
+                {
+                    Component = component,
+                    User = args.User,
+                },
+                TargetCancelledEvent = new SpoolingCancelEvent()
+            };
+
+            _popupSystem.PopupEntity(Loc.GetString("cassette-repair-start", ("item", component.Owner)), args.User, Filter.Entities(args.User));
+            _doAfterSystem.DoAfter(doAfterEventArgs);
+        }
+
+        private void OnSpoolingComplete(EntityUid uid, CassetteTapeComponent component, SpoolingCompleteEvent args)
+        {
+            _popupSystem.PopupEntity(Loc.GetString("cassette-repair-finish", ("item", component.Owner)), args.User, Filter.Entities(args.User));
+
+            _tagSystem.AddTag(component.Owner, "CassetteTape");
```

Comment:
The tag is for the tape recorders itemslot whitelist. It removes the tag when damaged because you can't insert an unspooled cassette into a player. It adds it back when repaired